validate.go

```diff
@@ -336,29 +336,55 @@ func usesFieldsBoundaryDirective(schema *ast.Schema) bool {
 
 // validateBoundaryFields checks that all boundary types have a getter and all getters are matching with a boundary type
 func validateBoundaryFields(schema *ast.Schema) error {
-	boundaryTypes := make(map[string]struct{})
+	type boundaryType struct {
+		hasIndividualGetter bool
+		hasArrayGetter      bool
```

Comment:
The graphql spec term is a `List` not an `Array` so we should be consistent with the type name here and throughout the code too.

So error messages, variable names should mention list instead.